autopush/router/apns2.py

```diff
@@ -36,13 +39,21 @@ class APNSException(Exception):
     pass
 
 
+class APNSConnection:
+    def __init__(self, conn):
+        self._conn = conn
+        self._last_used = time.time()
```

Comment:
It doesn't look like `_last_use` is ever modified. Maybe that should happen in `_get_connection`?

Reply:
Going to add commentary around this. APNSConnection is a wrapper for "idle" connections, so `_last_used` is only set when a connection is returned to the idle pool. When a connection becomes active again, it's removed from the `collections` set. This prevents `_reap` from ever force closing an active connection.